src/sagemaker/local/data.py

```diff
@@ -233,8 +236,29 @@ def split(self, file):
         Returns: generator for the individual records that were split from
         the file
         """
-        with open(file, "r") as f:
-            yield f.read()
+        with open(file, "rb") as f:
+            buf = f.read()
+            if not self._is_binary(buf):
+                buf = buf.decode()
+            yield buf
+
+    def _is_binary(self, buf):
+        """binary check.
+
+        binary or text check.
```

Comment:
doc comment should be clearer.  recommend:

```
Check whether `buf` contains binary data. 
Returns true if `buf` contains any non-utf-8 characters.

Args:
            buf (bytes): data to inspect

Returns:
           True if data is binary, otherwise False
```